versioning/src/main/scala/org/typelevel/sbt/TypelevelVersioningPlugin.scala

```diff
@@ -129,7 +131,10 @@ object TypelevelVersioningPlugin extends AutoPlugin {
         version += s"-$formatted"
       }
 
-      if (isSnapshot.value) version += "-SNAPSHOT"
+      val isPublishingToCentalPortal =
+        sonatypeCredentialHost.value == Sonatype.sonatypeCentralHost
+
+      if (isSnapshot.value && !isPublishingToCentalPortal) version += "-SNAPSHOT"
```

Comment:
Sonatype Central does not accept `-SNAPSHOT` versions.

> The Central Publisher Portal does not support -SNAPSHOT releases, and deployments of -SNAPSHOT releases will have a version cannot be a SNAPSHOT error in their validation results. Versions should only be published to the Portal if they are intended to reach Maven Central.

Reply:
Oh sorry, I missed the `!` :)

Anyway, I feel it's disingenous for `isSnapshot` to be `true` but lacking the `-SNAPSHOT`.

Reply:
Looking at how the setting is defined ...

https://github.com/sbt/sbt/blob/2f936295f158117baf9397ce8e9487bf9be8c87e/main/src/main/scala/sbt/Keys.scala#L520

Reply:
Mmm, also disingenuous if `isSnapshot` is true but it's not published to the snapshot repo?

https://s01.oss.sonatype.org/content/repositories/snapshots/